docs/glossary.md

```diff
@@ -0,0 +1,53 @@
+# Terminology
+
+### Jobs 
+
+Jobs, or Primary Jobs, are special types of tasks run in simulations.
+
+### Tasks
+
+Tasks are any thing run within simulations.
+
+### Taskflow
+
+Taskflows are a collection of tasks and jobs which are run for simulations.
+
+### Workflow
+
+Workflows are collections of taskflows which define how you run simulations and jobs.
+
+### Projects
+
+Projects are collections of simulations.
+
+### Simulations
+
+Simulations are collections of workflows.
+
+### Clusters
+
+Clusters are any machine that you can run tasks on.
+
+### Traditional Clusters
+
+Traditional clusters are machines that you have access and permissions for.
```

Comment:
Maybe a better definition would mention it corresponds to physical hardware and/or that fact it persists.